coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/util/GeoserverUtil.java

```
@@ -365,13 +366,17 @@ public static Service addRasterLayer(String geoServerEndpoint, InputStream zipFi
                         layerEncoder.setDefaultStyle(DEFAULT_RASTER_STYLE);
                 
                     // Geoserver Manager BUG Alert for v2.4 ...Creating a GeoTIFF coverage via REST works if the coverage's name is the same as the store name, but fails otherwise. Setting nativeName or nativeCoverageName does not help. Changing the name after creating the coverage works fine.    
-                    publisher.publishExternalGeoTIFF(PORTAL_WORKSPACE, fileName, unzippedFile, coverageEncoder, layerEncoder); // #TODO# what to do if the workspace:fileName is already in use?
-                                    
-                    log.info("Published GeoTiff!!!");
-                    log.info("In GeoserverUtil, about to add wmsService with layer name: " + layerId);
-                    Service rasterService = wmsService(layerId);
-                    log.info("Added layer to wms service.");
-                    
+                    RESTCoverageStore store = publisher.publishExternalGeoTIFF(PORTAL_WORKSPACE, fileName, unzippedFile, coverageEncoder, layerEncoder); // #TODO# what to do if the workspace:fileName is already in use?
+                    if (null == store) {
+                            //if store or layer creation failed
+                            log.info("Error publishing GeoTiff in GeoServer.");
+                            rasterService = null;
```

Here we are returning null. We could also throw an Exception instead.